var/spack/repos/builtin/packages/specfem3d-globe/package.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Specfem3dGlobe(AutotoolsPackage):
+    """Program specfem3D from SPECFEM3D_GLOBE is
+    a 3-D spectral-element solver for the Earth.
+    It uses a mesh generated by program meshfem3D."""
+
+    homepage = "https://github.com/geodynamics/specfem3d_globe"
+    url      = "https://github.com/geodynamics/specfem3d_globe/archive/v7.0.2.tar.gz"
+
+    version('7.0.2', sha256='78b4cfbe4e5121927ab82a8c2e821b65cdfff3e94d017303bf21af7805186d9b')
+
+    variant("cuda", default=False,
+            description="Build with CUDA code generator")
+    variant('opencl', default=False,
+            description="Build with OpenCL code generator")
+    variant('openmp', default=True,
+            description="Build with OpenMP code generator")
+    variant('double-precision', default=False,
+            description="Treat REAL as double precision")
+
+    depends_on('mpi')
+    depends_on('cuda', when='+cuda')
+    depends_on('opencl', when='+opencl')
+
+    # When building with the gcc compiler,'Werror' is added to FFLAGS.
+    # In the case of using the gcc compiler and the default simulation
+    # settings, there is the process which always causes
+    # array out-of-bounds reference error.
```

Comment:
:scream:

Reply:
😱 Version 8.0.0 seems to solve this problem, so can 7.0.2 be allowed to remove `-Werror`?
The developer has contacted me that this countermeasure is possible:
https://github.com/geodynamics/specfem3d_globe/issues/717